src/pages/wallet-overview/wallet-overview.spec.ts

```diff
@@ -514,5 +456,493 @@ describe("WalletOverviewPage", () => {
+
+    let iosPromise = comp.writeFileMobile("location", "filename", "keystoredata", {replace: true}, "ios").then(data => {
+      expect(comp.writeFileMobile).toHaveBeenCalledWith("location", "filename", "keystoredata", {replace: true}, "ios");
+      expect(comp.showToastMessage).toHaveBeenCalledWith(jasmine.any(String), 2000, "bottom");
+      expect(comp.translations.get).toHaveBeenCalledWith("wallet_overview.saved_keystore_ios");
+    });
+
+    let otherPromise = comp.writeFileMobile("location", "filename", "keystoredata", {replace: true}, "android").then(data => {
+      expect(comp.translations.get).toHaveBeenCalledWith("wallet_overview.saved_keystore_android");
+    });
+    Promise.all([iosPromise, otherPromise]).then(data => {
+      done();
+    });
+  });
+
+  it("should present a toast with a close button when the file could not be written", (done) => {
+    let toast = new MockToast();
+
+    spyOn(fileNative, "writeFile").and.returnValue(Promise.reject(""));
+    spyOn(toastController, "create").and.returnValue(toast);
+    spyOn(toast, "present");
+
+    comp.writeFileMobile("location", "filename", "keystoredata", {replace: true}, "android").then(data => {
+      expect(toastController.create).toHaveBeenCalledWith({
+        message: jasmine.any(String),
+        position: "bottom",
+        showCloseButton: true,
+        closeButtonText: "Ok"
+      });
+      expect(toast.present).toHaveBeenCalled();
+      done();
+    });
+  });
+
+  it("should show that there are no transactions if it returns an empty array", (done) => {
+    spyOn(transactionHistoryService, "getTransactionHistory").and.returnValue(Promise.resolve([]));
+
+    comp.getTransactionHistory("").then(data => {
+      expect(comp.noTransactionHistoryVisibility).toBe("shown");
+      expect(comp.transactionHistoryVisibility).toBe("hidden");
+      done();
+    });
+  });
+
+  it("should show an alert and retry button when the transaction history could not be retrieved", (done) => {
+    let alert = new MockAlert();
+
+    spyOn(transactionHistoryService, "getTransactionHistory").and.returnValue(Promise.reject([]));
+    spyOn(alertController, "create").and.returnValue(alert);
+    spyOn(alert, "present");
+
+    comp.getTransactionHistory("").then(data => {
+      expect(alertController.create).toHaveBeenCalled();
+      expect(alert.present).toHaveBeenCalled();
+      done();
+    });
+  });
+
+  it("should show a retry modal for the exchanges when the request returns an empty OK response", (done) => {
+    spyOn(exchangeService, "getAvailableExchanges").and.returnValue(Promise.resolve([]));
+    spyOn(comp, "getAvailableExchanges").and.callThrough();
+    spyOn(comp, "presentExchangesRetryButton");
+
+    comp.getAvailableExchanges().then(data => {
+      expect(comp.presentExchangesRetryButton).toHaveBeenCalled();
+      done();
+    });
+  });
+
+  it("should push zero balances for XSM and XSP when the JSON is null or has no keys at all", (done) => {
```

Comment:
This test tests the null case but not the no 'no keys' case. Should this be added?